ChessBoard.UnitTests/ChessBoardGenerator.Tests.cs

```diff
@@ -0,0 +1,38 @@
+﻿using Moq;
+using Serilog;
+
+
+
+namespace ChessBoardApp.ChessBoard.Tests
+{
+
+
+    public class ChessBoardGeneratorTests
+    {
+
+        [Theory]
+        [InlineData("1\r\n",1,1)]
+        [InlineData("101\r\n010\r\n101\r\n010\r\n", 3,4)]
+        [InlineData("1010\r\n0101\r\n1010\r\n0101\r\n", 4,4)]
+        [InlineData("1010\r\n0101\r\n1010\r\n0101\r\n1010\r\n0101\r\n", 4,6)]
+        public void GenerateRectangleBoard_ShouldReturnCorrectChessBoard(string expected, int width, int length)
```

Comment:
Maybe moving chessBoardGenerator instance? 
```
public class ChessBoardGeneratorTests
    {
        private readonly Mock<ILogger> loggerMock;
        private readonly ChessBoardGenerator chessBoardGenerator;

        public ChessBoardGeneratorTests()
        {
            loggerMock = new Mock<ILogger>();
            chessBoardGenerator = new ChessBoardGenerator(loggerMock.Object);
        }

        [Theory]
        [InlineData("1\r\n", 1, 1)]
        [InlineData("101\r\n010\r\n101\r\n010\r\n", 3, 4)]
        [InlineData("1010\r\n0101\r\n1010\r\n0101\r\n", 4, 4)]
        [InlineData("1010\r\n0101\r\n1010\r\n0101\r\n1010\r\n0101\r\n", 4, 6)]
        public void GenerateRectangleBoard_ShouldReturnCorrectChessBoard(string expected, int width, int length)
        {
            // Act
            var result = chessBoardGenerator.GenerateRectangleBoard(width, length);

            // Assert
            Assert.Contains(expected, result);
        }
```
I also created a test for both OneLoop and TwoLoops:

Let me know if I should get rid of any of them and also if moving the chessBoardGenerator instance to private field within class was correct option

Reply:
Good job,  now something to keep in mind - internal and private fields are prefixed by "_" by convention, let's uptade this with next step. Good for now.
https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/coding-conventions#camel-case
also, since we're not using loggerMock anywhere but in constructor, it could now be made into local variable within  that constructor:
![image](https://github.com/Pauliukas7/dotnet-console-app/assets/2812503/f9d1ed99-dde8-4e33-be0b-bb0776db3ba1)

Resolved.